allauth/socialaccount/providers/yandex/provider.py

```diff
@@ -7,6 +7,9 @@
 
 class YandexAccout(ProviderAccount):
     def to_str(self):
+        default_email = self.account.extra_data.get("default_email")
+        if default_email:
+            return default_email
         first_name = self.account.extra_data.get("first_name", "")
         last_name = self.account.extra_data.get("last_name", "")
         name = " ".join([first_name, last_name]).strip()
```

Comment:
This is not Yandex specific, just rely on the base implementation.

Reply:
The base `to_str` does not look at `default_email`, are you saying it should?

Reply:
Yes, why not? Why add `full_name`, `first/last_name` lookups to a few providers requiring a custom provider specific `to_str()`  -- why not do this globally?

Reply:
Correction: not `default_email`, but definitely `first_name` and `last_name`.

Reply:
So to be precise: `if default_email` then return that, else just invoke `to_str()` which should take care of `first/last_name`.

Reply:
Done.